### PR TITLE
Added OpenAPI response structures for sys endpoints

### PR DESCRIPTION
**Pull request by lursu**
adding response objects for the sys endpoints laid out in [VAULT-12143](https://hashicorp.atlassian.net/browse/VAULT-12143?atlOrigin=eyJpIjoiOTNhMTI1MGFmOWFlNGZjZWE0NDk2MWRiOGM2YjkxY2YiLCJwIjoiaiJ9).

[VAULT-12143]: https://hashicorp.atlassian.net/browse/VAULT-12143?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: vault/logical_system_paths.go
Comment (AnPucel) [inline vault/logical_system_paths.go:2636]: I believe with dynamic keys we just added an empty `FieldSchema`

File: vault/logical_system_paths.go
Comment (averche) [inline vault/logical_system_paths.go:317]: I think most of the `"Update"` & `"Delete"` statuses should be 204 (`http.StatusNoContent`). If the response handler returns `nil, nil`, it will be translated into a 204 by the routing logic. The ones with data and/or warnings will be returned as 200s.

Comment (lursu) [reply vault/logical_system_paths.go:317]: done

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go]: removed all occurrences.

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:1736]: this is the old way for how vault did responses, but I dont know what the fields of the actual response would be

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:1771]: looks like this is no content
https://github.com/hashicorp/vault/blob/901876f27abc9e74904aaebaba7a31bc9cb970a6/vault/logical_system.go#L3771

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:2055]: seems like `handlePoliciesRead` returns a differently shaped response
https://github.com/hashicorp/vault/blob/901876f27abc9e74904aaebaba7a31bc9cb970a6/vault/logical_system.go#L2867-L2879

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:2191]:
```suggestion
									Type: framework.TypeStringSlice,
									Required: false
```
* the suggestion may cause go fmt errors

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:2439]:
```suggestion
									Type:        framework.TypeInt,
```
looks like this is cast to int
https://github.com/hashicorp/vault/blob/901876f27abc9e74904aaebaba7a31bc9cb970a6/vault/logical_system.go#L1595

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:2449]:
```suggestion
									Type:        framework.TypeInt,
```
looks like this is cast to int
https://github.com/hashicorp/vault/blob/901876f27abc9e74904aaebaba7a31bc9cb970a6/vault/logical_system.go#L1594

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:2472]: where do you see the field `allowed_request_headers`?

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:2478]: general question: should we be using `TypeKVPairs` or `TypeMap`? (if there's a difference)

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:2530]: might have to test this, seems like the function returns a response struct, forget if that means 200 or 204

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:2686]: looks like no content on success
https://github.com/hashicorp/vault/blob/901876f27abc9e74904aaebaba7a31bc9cb970a6/vault/logical_system.go#L1338

File: vault/login_mfa.go
Comment (dhuckins) [inline vault/login_mfa.go:94]: seems like `error` would only show up on non-200 responses. otherwise an empty response

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:826]: So when actually reading through the code of `logical.ErrorResponse` when there is an error it doesn't give it a non 200 status code. So as far as I understood it adds an error field to 200's. Which probably isn't the best if true but that seems like another pr.
https://github.com/hashicorp/vault/blob/31772c930b5c782b0879b2107332c715867d0473/sdk/logical/response.go#L124-L133

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:1392]: Yeah but given goes zero value for time if the key is set something will always been sent, so this may actually even be a required field? @antonakv do you have any further insights here?

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:2051]: Ah looks like copy pasta error, fixed.

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:2468]: should have been `"passthrough_request_headers"` fixed now.

Comment (lursu) [reply vault/logical_system_paths.go:2468]: ah nvm that was already there... yeah idk how I added that but removed now.

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:2474]: not sure, but the request type for this field is `TypeKVPairs` so I kept it the same.

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:2526]: probably just needs a `200` i updated it.

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:2682]: Same issue as other error resp.

File: vault/login_mfa.go
Comment (lursu) [reply vault/login_mfa.go:94]: same point as above.

Comment (dhuckins) [reply vault/login_mfa.go:94]: nice catch, might need to update my PR's

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:915]:
```suggestion
									Required:    false,
```
`logical.ListResponse` only has `keys` when there's more than one item
https://github.com/hashicorp/vault/blob/2267b042dba5ebe278867662ea059c4bda17356d/sdk/logical/response.go#L140-L142

_but_ would think we always have at least 1 plugin... 🤷

File: vault/logical_system_paths.go
Comment (dhuckins) [inline vault/logical_system_paths.go:1345]: `logical.ListResponse` only has `keys` when there's more than one item
https://github.com/hashicorp/vault/blob/2267b042dba5ebe278867662ea059c4bda17356d/sdk/logical/response.go#L140-L142
```suggestion
									Required:    false,
```

File: vault/logical_system_paths.go
Comment (dhuckins) [reply vault/logical_system_paths.go:1732]: cc @averche

**Review by dhuckins — APPROVED** [review]
think there's a few questions @averche should take a look at but lgtm

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:992]: Ok so I should just remove these then?

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:915]: Ah good catch.

File: vault/logical_system_paths.go
Comment (averche) [inline vault/logical_system_paths.go:2023]: Same as above.

File: vault/logical_system_test.go
Comment (averche) [inline vault/logical_system_test.go:5148]: If it's not too much work, could you please switch these to `GetResponseSchema(t, ...b.Route(...))` to make them less brittle. See https://github.com/hashicorp/vault/pull/18865 for a better example.

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go:1429]: .... oops

File: vault/logical_system_paths.go
Comment (lursu) [reply vault/logical_system_paths.go]: sorry is actually `"key_info"` not keys... related line from handler. https://github.com/hashicorp/vault/blob/b741fa893d775a252c5e518c015aa1c94e30cc63/vault/logical_system.go#L2840

File: vault/logical_system_test.go
Comment (lursu) [reply vault/logical_system_test.go]: Ah I didn't know that was possible much much better and I updated everywhere.

File: vault/logical_system_paths.go
Comment (averche) [reply vault/logical_system_paths.go:1429]:
```suggestion
									Required:    true,
```

File: vault/logical_system_paths.go
Comment (averche) [reply vault/logical_system_paths.go:1434]:
```suggestion
									Required:    true,
```

File: vault/logical_system_paths.go
Comment (averche) [reply vault/logical_system_paths.go]: The code you referenced seems to only apply to `PolicyTypeEGP` whereas the function deals with `PolicyTypeACL`.

Comment (lursu) [reply vault/logical_system_paths.go]: Oh dang good catch.

File: vault/logical_system_paths.go
Comment (averche) [inline vault/logical_system_paths.go:2103]:
```suggestion
```

**Review by averche — DISMISSED** [review]
LGTM

**Review by averche — COMMENTED** [review]
Please update other tests for the affected functionality (e.g. https://github.com/hashicorp/vault/blob/main/vault/external_tests/mfa/login_mfa_test.go for sys/mfa)

**Review by dhuckins — APPROVED** [review]

**Review by averche — APPROVED** [review]
LGTM

Before merging, please adjust the title & PR description (references to the RFC + related PR's would be great)

**Pull request merged**
